lib/commands/exam.js

```diff
@@ -239,14 +242,24 @@ module.exports = TestCommand.extend({
     const testemEvents = this.testemEvents;
     const ui = this.ui;
     const events = config.custom_browser_socket_events || {};
-    const testExecutionFileName = `test-execution-${Date.now()}.json`;
-    const browserExitHandler = function() {
+
+    const browserExitHandler = function(failed = false) {
+      if (!failed && commands.get('loadBalance')) {
+        try {
+          const browserId = getBrowserId(this.launcher.settings.test_page);
+          log.info(`Browser ${browserId} exiting. [ # of modules in current module queue ${testemEvents.stateManager.getTestModuleQueue().length} ]`);
+        } catch(err) {
+          const errMsg = `A browser Id is not available: ${err.message}`;
```

Comment:
Several notes about re-throwing errors with additional data.

1. `err` from catch can by anything, null, undefined, a string, etc, so we should check that before accessing `.message`
2. the method of error wrapping may loose important information, let me instead recommend we simply augment the existing message.

```js

const message = `[ # of modules in current module queue ${testemEvents.stateManager.getTestModuleQueue().length} ]` 
 if (typeof err === 'object' && err !== null) {
  // the err thrown can have its message augmented, let's augment it, but leave the original error as it. This ensures we do not lose important error context.
  err.message = `${err.message} \n ${message}`;
} else {
  // we are in a catch, but the `err` doesn't appear to be a real error and since we do want to include our message, we create a new error with said message and throw it.
  throw new Error(message);
}

throw err;
```